### DISCUSSION TITLE
option to enable void trading

### DISCUSSION THREAD

**Discussion by willymv** [PurpurMC/Purpur#846] [__Completed__] [answered]
Currently 'void trading' or being able to continuously trade with a villager through some End Gateway trickery is patched out in all forks of Paper (including Purpur). While this is considered an exploit/bug in Minecraft, the technical community who's gameplay revolves around quirks of Java Edition like this would probably appreciate it at least being configurable to re-enable it.

Video that showcases the bug: https://youtu.be/zhsSW1q3t2I?t=510

The video above shows how it works in Vanilla, however on Paper (and all the forks of it) the Villager GUI disappears when the chunks unload. If any trades are made before the GUI disappears, they are treated as regular trades and not 'void' trades.

### Describe the solution you'd like.
A configurable option in the purpur.yml file to enable Void Trading (default not enabled).

### Describe alternatives you've considered.
Currently this exploit/bug works on Fabric and Spigot servers. If I could simply switch to Fabric or Spigot, I would, however this is not an option for myself and I assume others who may be interested in using this exploit/bug in their gameplay.

### Agreements
Please agree to the following.

- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] I have ensured the feature I'm requesting isn't already in the latest supported Purpur build.

### Other
I understand that this was a patch that Paper added and that many players may think this bug ruins the game and should stay patched out, however I came here because I have seen Purpur add configurable options for exploit/bug patches in the past and assumed that there'd be more of a chance this wouldn't immedietly be thrown out here.

**Comment by BillyGalbreath** [comment, marked as answer]
~~Depends on how they patched it out. Do you have details to point me in that direction so I can see what they did?~~

Nevermind, I found it.

Implemented in build 1505

https://github.com/PurpurMC/Purpur/commit/b27699e339325e637d5ff737974de6d13f16a575

**Reply by willymv** [reply]
Thank you, you're a legend!
